token/services/selector/sherdlock/service.go

```diff
@@ -17,17 +18,23 @@ import (
 	"github.com/pkg/errors"
 )
 
-const retrySelectionBackoff = 5 * time.Second
-
 type SelectorService struct {
 	managerLazyCache utils.LazyProvider[*token.ManagementService, token.SelectorManager]
 }
 
-func NewService(tokenDBManager *tokendb.Manager, tokenLockDBManager *tokenlockdb.Manager, metricsProvider metrics.Provider) *SelectorService {
+func NewService(tokenDBManager *tokendb.Manager, tokenLockDBManager *tokenlockdb.Manager, cfg driver.ConfigService, metricsProvider metrics.Provider) *SelectorService {
+	c := SelectorConfig{}
+	err := cfg.UnmarshalKey("token.selector", &c)
+	if err != nil {
+		panic("invalid config for key [token.selector]: expect retryInterval (duration) and numRetries (integer))")
```

Comment:
do you think panicing here is the right way to handle the error? Can we unmarshall outside the function and pass directly SelectorConfig?